src/amm/mod.rs

```diff
@@ -19,46 +21,53 @@ use self::{erc_4626::ERC4626Vault, uniswap_v2::UniswapV2Pool, uniswap_v3::Uniswa
 #[async_trait]
 pub trait AutomatedMarketMaker {
     /// Returns the address of the AMM.
-    fn address(&self) -> H160;
+    fn address(&self) -> Address;
 
     /// Syncs the AMM data on chain via batched static calls.
-    async fn sync<M: Middleware>(&mut self, middleware: Arc<M>) -> Result<(), AMMError<M>>;
+    async fn sync<T: Transport + Clone, P: Provider<T, AnyNetwork>>(
```

Comment:
Lets update to use `where` clauses instead. 

Also can we update to use `N: Network` instead of `AnyNetwork`? Lmk if there is a reason you went with `AnyNetwork` instead.